examples/c/ksyscall.bpf.c

```diff
@@ -0,0 +1,50 @@
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_core_read.h>
+
+#define TASK_COMM_LEN 16
+
+SEC("ksyscall/tgkill")
+int BPF_KSYSCALL(tgkill_entry, pid_t tgid, pid_t tid, int sig) 
+{
+
+    char comm[TASK_COMM_LEN];
+    __u32 caller_pid = bpf_get_current_pid_tgid() >> 32;
+	if (sig == 0) {
+		/**
+			If sig is 0, then no signal is sent, but existence and permission
+			checks are still performed; this can be used to check for the
+			existence of a process ID or process group ID that the caller is
+			permitted to signal.
+		*/
+		return 0;
+	}
+
+	bpf_get_current_comm(&comm, sizeof(comm));
+    bpf_printk("tgkill syscall called by PID %d (%s) for thread id %d with signal %d.", caller_pid, comm, tid, sig);
+	return 0;
+}
+
+SEC("ksyscall/kill")
+int BPF_KSYSCALL(entry_probe, pid_t pid, int sig) 
+{
+
+    char comm[TASK_COMM_LEN];
+    __u32 caller_pid = bpf_get_current_pid_tgid() >> 32;
+	if (sig == 0) {
+		/**
```

Comment:
no need to use `/**`, this is not a doc comment, just `/*`. It's also kernel coding style to use the following style:

```
/* Start of
 * multi-line
 * comment
 */
```

let's keep consistent styling